### PR TITLE
Add auto agent

### PR DESCRIPTION
**Pull request by nqyy**

### CODE REVIEW

**Review by sebbyjp — COMMENTED** [review]
Needs functools.wraps to wrap attributes it wraps. Also should have a method to print the available tasks. The dispatching should happen at both init and call time. Use a literal to give type hints for tasks and | str for the instruction arg. Interface should be identical to LanguageAgent so either rename to instruction or language to task. The fallback should also be to route to a languageagent if no task was matched

**Comment by nqyy** [comment]
> Needs functools.wraps to wrap attributes it wraps. Also should have a method to print the available tasks. The dispatching should happen at both init and call time. Use a literal to give type hints for tasks and | str for the instruction arg. Interface should be identical to LanguageAgent so either rename to instruction or language to task. The fallback should also be to route to a languageagent if no task was matched

Updated the changes needed as discussed.

File: README.md [hunk 1]
```diff
@@ -159,6 +161,8 @@ _Embodied Agents are not yet capable of learning from in-context experience_:
 - [x] OpenVLA Motor Agent
 - [x] Automatic dataset recording on Robot
 - [x] Yolo, SAM2, DepthAnything Sensory Agents
+- [x] Auto Agent
+- [ ] Central Nervous System (CNS)
```
Comment (sebbyjp) [inline README.md:165]:
```suggestion
```
Remove this

File: README.md [hunk 2]
```diff
@@ -261,9 +265,7 @@ To learn more about all of the possibilities with embodied agents, check out the
 - You can `pack` a list of `Sample`s or Dicts into a single `Sample` or `Dict` and `unpack` accordingly?
 - You can `unflatten` any python structure into a `Sample` class so long you provide it with a valid json schema?
 
-<details> <summary><h2 style="display: inline-block;">Deep Dive</h2></summary>
-
-## Building Blocks
+## Deep Dive
```
Comment (sebbyjp) [inline README.md:268]:
```suggestion
## API Reference
```

File: README.md [hunk 3]
```diff
@@ -81,6 +81,8 @@ This repository is broken down into 3 main components: **Agents**, **Data**, and
 - **Motor Agents** always return a `Motion`.
 - **Sensory Agents** always return a `SensorReading`.
 
+For convenience, we also provide **AutoAgent** which dynamically initializes the right agent for the specified task. See [Deep Dive section](#auto-agent) below for more.
```
Comment (sebbyjp) [inline README.md:84]:
```suggestion
For convenience, we also provide **AutoAgent** which dynamically initializes the right agent for the specified task. See [API Reference](#auto-agent) below for more.
```

**Review by sebbyjp — APPROVED** [review]

**Review by sebbyjp — APPROVED** [review]

**Review by Zaid-Waghoo — APPROVED** [review]

**Pull request merged**